datumaro/util/annotation_util.py

```diff
@@ -178,11 +188,18 @@ def OKS(a, b, sigma=None, bbox=None, scale=None):
     dists = np.linalg.norm(p1 - p2, axis=1)
     return np.sum(np.exp(-(dists ** 2) / (2 * scale * (2 * sigma) ** 2)))
 
-def smooth_line(points, segments):
-    assert 2 <= len(points) // 2 and len(points) % 2 == 0
+def approximate_line(points: np.ndarray, segments: int) -> np.ndarray:
+    """
+    Approximates a 2d line to the required number of segments. Points are
+    distributed uniformly across the resulting line.
 
-    if len(points) // 2 == segments:
-        return points
+    Returns:
+      new_points (ndarray): a flat array of new line point coordinates.
+        The shape is [segments, 2], the layout is [[x0, y0], [x1, y1], ...].
```

Comment:
This description seems contradictory - if the array has two dimensions, it's not flat.

Also, the input array should be described as well.

Reply:
Fixed